lib/services/ngsi/subscriptionService.js

```diff
@@ -131,18 +186,93 @@ function subscribe(device, triggers, content, callback) {
             config.getConfig().contextBroker.port + '/v1/subscribeContext';
     }
 
-    request(options, createSubscriptionHandler(device, triggers, store, callback));
+    request(options, createSubscriptionHandlerNgsi1(device, triggers, store, callback));
 }
 
 /**
- * Generate a new unsubscription request handler, based on the device and subscription ID given to the function.
+ * Makes a subscription for the given device's entity using NGSIv2, triggered by the given attributes.
+ * The contents of the notification can be selected using the "content" array (that can be left blank
+ * to notify the complete entity).
+ *
+ * @param {Object} device       Object containing all the information about a particular device.
+ * @param {Object} triggers     Array with the names of the attributes that would trigger the subscription
+ * @param {Object} content      Array with the names of the attributes to retrieve in the notification.
+ */
+function subscribeNgsi2(device, triggers, content, callback) {
+    var options = {
+        method: 'POST',
+        headers: {
+            'fiware-service': device.service,
+            'fiware-servicepath': device.subservice
+        },
+        json: {
+                subject:
+                {
+                    entities: [
+                        {
+                            id: device.name,
+                            type: device.type
+                        }
+                    ],
+
+                    condition: {
+                        attrs: triggers
+                    },
+                },
+                notification: {
+                    http: {
+                        url: config.getConfig().providerUrl + '/notify'
+                    },
+                    attrs: content || []
+                },
+                expires: config.getConfig().deviceRegistrationDuration,
```

Comment:
Looking to the NGSIv1 version of the code I understand that deviceRegistrationDuration could be ommited, so there is a default option for "permanente" after a ||

```
config.getConfig().deviceRegistrationDuration || 'P100Y'
```

In NGSIv2 case, given that the concept of permanent subscriptions exict, I understand you should manage to omit `expires` field in the case deviceRegistrationDuration is not used.

Reply:
Fixed in 4232cc6baf524d74560dda9ed23f9c044286954b